libbeat/common/kubernetes/metadata/node.go

```diff
@@ -46,6 +48,7 @@ func (n *node) Generate(obj kubernetes.Resource, opts ...FieldOptions) common.Ma
 
 	meta := n.resource.Generate("node", obj, opts...)
 	// TODO: Add extra fields in here if need be
+	meta.Put("node.hostname", n.hostname)
```

Comment:
Shouldn't we use the hostname in `obj.Status.Addresses[...].Address`? I am concerned that when using cluster scope we can be setting here the hostname of the observer, and not the one the event belongs to.

Reply:
++ good catch!